src/ConnectionAutomaton.cpp

```diff
@@ -109,10 +121,14 @@ void ConnectionAutomaton::closeDuplexConnection(folly::exception_wrapper ex) {
     connectionOutput_.onComplete();
   }
   connectionInputSub_.cancel();
-  connection_.reset();
+}
 
-  stats_.socketClosed();
-  onClosed_();
+void ConnectionAutomaton::closeWithError(Frame_ERROR&& error) {
+  VLOG(4) << "closeWithError "
+          << error.payload_.data->cloneAsValue().moveToFbString();
+
+  outputFrameOrEnqueue(error.serializeOut());
+  close(folly::exception_wrapper(), StreamCompletionSignal::ERROR);
```

Comment:
Yeah, we have been using exception_wrapper() (default ctor) to indicate no error. This could be found on multiple places in our codebase and also in other FB code.